prov/util/src/util_mr_map.c

```diff
@@ -82,6 +83,8 @@ int ofi_mr_map_insert(struct ofi_mr_map *map, const struct fi_mr_attr *attr,
 	if (map->mode & FI_MR_PROV_KEY)
 		item->requested_key = map->key++;
 
+	key_is_optional = !(attr->access & (FI_REMOTE_READ | FI_REMOTE_WRITE));
+
 	ret = ofi_rbmap_insert(map->rbtree, &item->requested_key, item, NULL);
```

Comment:
The first insertion will succeed because the key is unique. Meaning `fi_mr_key` will return the `requested_key` from mr_attr for the first region and `FI_KEY_NOTAVAIL` for the others.

Also, if there is a future `fi_mr_reg` call for the same region with different access flags, they will overlap in the rb tree.

I think we should not insert into the rb tree if `key_is_optional` is true.

Reply:
Updated.